core/src/test/java/io/grpc/PickFirstLoadBalancer2Test.java

```diff
@@ -212,13 +214,31 @@ public void pickAfterStateChangeAfterResolution() throws Exception {
 
   @Test
   public void nameResolutionError() throws Exception {
-    loadBalancer.handleNameResolutionError(Status.NOT_FOUND.withDescription("nameResolutionError"));
+    Status error = Status.NOT_FOUND.withDescription("nameResolutionError");
+    loadBalancer.handleNameResolutionError(error);
     verify(mockHelper).updatePicker(pickerCaptor.capture());
     PickResult pickResult = pickerCaptor.getValue().pickSubchannel(Attributes.EMPTY,
         new Metadata());
     assertEquals(null, pickResult.getSubchannel());
-    assertEquals(Status.NOT_FOUND.getCode(), pickResult.getStatus().getCode());
-    assertEquals("nameResolutionError", pickResult.getStatus().getDescription());
+    assertEquals(error, pickResult.getStatus());
+    verifyNoMoreInteractions(mockHelper);
+  }
+
+  @Test
+  public void nameResolutionSuccessAfterError() throws Exception {
+    InOrder inOrder = inOrder(mockHelper);
+
+    loadBalancer.handleNameResolutionError(Status.NOT_FOUND.withDescription("nameResolutionError"));
+    inOrder.verify(mockHelper).updatePicker(any(Picker.class));
+
+    loadBalancer.handleResolvedAddresses(servers, affinity);
+    inOrder.verify(mockHelper).createSubchannel(eq(new EquivalentAddressGroup(socketAddresses)),
+        eq(Attributes.EMPTY));
+    inOrder.verify(mockHelper).updatePicker(pickerCaptor.capture());
+
+    assertEquals(pickerCaptor.getValue().pickSubchannel(Attributes.EMPTY, new Metadata()),
```

Comment:
You probably also want to verify that `pickSubchannel()` actually returns `subchannel`.